modules/mailchimp_ecommerce_commerce/mailchimp_ecommerce_commerce.module

```diff
@@ -296,8 +297,16 @@ function mailchimp_ecommerce_commerce_commerce_cart_product_add($order, $product
 
   // If this order is new, that means this is the first line item;
   // send the entire cart to MailChimp first.
-  if ($order->created == $order->changed) {
-    _mailchimp_ecommerce_commerce_send_cart($order);
+  if (empty($order->data['mc_id'])) {
+    // @todo: Do this without a wrapper.
+    $order_wrapper = entity_metadata_wrapper('commerce_order', $order);
+    $mc_order = _mailchimp_ecommerce_commerce_build_order($order_wrapper);
+    $added = mailchimp_ecommerce_add_cart($order->order_id, $mc_order['customer'], $mc_order['order_data']);
+    if (!empty($added)) {
+      $order->data['mc_id'] = time();
+      commerce_order_save($order);
```

Comment:
This is a local order save yes? What's this bit do?

Reply:
Yup, that's a "flag" I'm setting in order data, that lets Commerce know whether an order has been added to the MailChimp API. When an order gets deleted from MC API (for instance, due to removing the _last_ line item from a shopping cart), we also delete that "flag".

This is the most effective way I've found of making sure we don't try calling `update_cart_line_process` if a cart doesn't even exist in MCAPI yet. If the order doesn't have the flag, we can assume the order doesn't exist in MailChimp, and in that case, send the entire thing (or queue it to be sent).

My reasoning is that the API is somewhat dumb, as it should be, and since we have no real way of knowing _which_ platform or system is calling the lower-level API methods like `update_cart_line_process()` then we probably need to figure that out higher up in the chain, as I've done here.